File: unitri-ppi/aulas-spring/src/main/java/com/example/aulasspring/service/VeiculoService.java

package com.example.aulasspring.service;

import com.example.aulasspring.model.Veiculo;
import com.example.aulasspring.repository.VeiculoRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class VeiculoService {

    @Autowired
    private VeiculoRepository repository;

    public List<Veiculo> findAll() {
        return repository.findAll();
    }

    public Veiculo findOne(Long id) {
        return repository.getOne(id);
    }

    public Veiculo save(Veiculo vei) {
        return repository.saveAndFlush(vei);
    }

    public void delete(Long id) {
        repository.deleteById(id);
    }
}

File: unitri-ppi/aula-07/src/main/kotlin/com/example/aula07/service/VeiculoService.kt
package com.example.aula07.service

import com.example.aula07.model.Veiculo
import com.example.aula07.repository.VeiculoRepository
import org.springframework.stereotype.Service


@Service
class VeiculoService(
    val repository: VeiculoRepository
) {

    fun findAll(): List<Veiculo> = repository.findAll()

    fun findOne(id: Long): Veiculo = repository.getOne(id)

    fun save(post: Veiculo): Veiculo = repository.saveAndFlush(post)

    fun delete(id: Long) = repository.deleteById(id)
}